File: HomeworkManager/HomeworkManager.Model/CustomEntities/User/UserListRow.cs
namespace HomeworkManager.Model.CustomEntities.User;

public class UserListRow
{
    public required Guid UserId { get; set; }
    public required string FullName { get; set; }
    public required string Username { get; set; }
    public required string Email { get; set; }
    public string Roles { get; set; } = string.Empty;
}

File: HomeworkManager.Angular/ClientApp/src/app/shared-module/models/user/user-list-row.ts
export class UserListRow {
  userId!: string
  fullName!: string
  username!: string
  email!: string
  roles!: string

  static compareFullName(user1: UserListRow, user2: UserListRow) {
    if (user1.fullName < user2.fullName) {
      return -1;
    } else if (user1.fullName == user2.fullName) {
      return 0;
    } else {
      return 1
    }
  }

  static displayFn(user: UserListRow): string {
    return user
      ? user.fullName + '(' + user.email + ')'
      : '';
  }
}